### PR TITLE
New package: i3lock-color-2.11

### PR DESCRIPTION
**Pull request by jtalowell**

### CODE REVIEW

File: srcpkgs/i3lock-color/template [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'i3lock-color'
+pkgname=i3lock-color
+version=2.11
+revision=1
+wrksrc="${pkgname}-${version}-c"
+only_for_archs="i686 x86_64"
```
Comment (maxice8) [inline srcpkgs/i3lock-color/template]: why is this such ? it certainly doesn't seem like it

File: srcpkgs/i3lock-color/template [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+build_style=gnu-configure
+conf_files="/etc/pam.d/i3lock"
+configure_args="--prefix=${DESTDIR}/usr/ --sysconfdir=${DESTDIR}/etc/"
+makedepends="pkg-config automake autoconf libev-devel cairo-devel pam-devel libxkbcommon-devel xcb-util-devel xcb-util-image-devel libjpeg-turbo-devel"
+depends="libev-devel cairo-devel pam-devel libxkbcommon-devel xcb-util-devel xcb-util-image-devel libjpeg-turbo-devel"
```
Comment (maxice8) [inline srcpkgs/i3lock-color/template]: remove the depends=

File: srcpkgs/i3lock-color/template [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'i3lock-color'
+pkgname=i3lock-color
+version=2.11
+revision=1
+wrksrc="${pkgname}-${version}-c"
+only_for_archs="i686 x86_64"
+build_style=gnu-configure
+conf_files="/etc/pam.d/i3lock"
+configure_args="--prefix=${DESTDIR}/usr/ --sysconfdir=${DESTDIR}/etc/"
+makedepends="pkg-config automake autoconf libev-devel cairo-devel pam-devel libxkbcommon-devel xcb-util-devel xcb-util-image-devel libjpeg-turbo-devel"
```
Comment (maxice8) [inline srcpkgs/i3lock-color/template]: move pkg-config and automake to hostmakedepends=

File: srcpkgs/i3lock-color/template [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'i3lock-color'
+pkgname=i3lock-color
+version=2.11
+revision=1
+wrksrc="${pkgname}-${version}-c"
+only_for_archs="i686 x86_64"
+build_style=gnu-configure
+conf_files="/etc/pam.d/i3lock"
+configure_args="--prefix=${DESTDIR}/usr/ --sysconfdir=${DESTDIR}/etc/"
```
Comment (maxice8) [inline srcpkgs/i3lock-color/template]: superfluous, already set

File: srcpkgs/i3lock-color/template [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+makedepends="pkg-config automake autoconf libev-devel cairo-devel pam-devel libxkbcommon-devel xcb-util-devel xcb-util-image-devel libjpeg-turbo-devel"
+depends="libev-devel cairo-devel pam-devel libxkbcommon-devel xcb-util-devel xcb-util-image-devel libjpeg-turbo-devel"
+conflicts="i3lock"
```
Comment (maxice8) [inline srcpkgs/i3lock-color/template]: put at the end of the variables=

File: srcpkgs/i3lock-color/template [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'i3lock-color'
+pkgname=i3lock-color
+version=2.11
+revision=1
+wrksrc="${pkgname}-${version}-c"
+only_for_archs="i686 x86_64"
+build_style=gnu-configure
+conf_files="/etc/pam.d/i3lock"
```
Comment (maxice8) [inline srcpkgs/i3lock-color/template]: put at the end of variables

File: srcpkgs/i3lock-color/template (hunk 1 above)
Comment (jtalowell) [reply srcpkgs/i3lock-color/template]: I've only built and tested the package on x86_64. I noticed the AUR package for this had it listed as only being for the archs I listed above which is why I had it. 

I've removed the line.

Comment (maxice8) [reply srcpkgs/i3lock-color/template]: Arch Linux just builds for x86_64, and on occasion where 32bit packages for x86_64 environments benefit (like wine) i686.

Void Linux builds for x86_64, i686, armv[56], aarch64.

only_for_arches should be used in occasions where like upstream doesn't support that architecture and it is broken.

File: srcpkgs/i3lock-color/template [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'i3lock-color'
+pkgname=i3lock-color
+version=2.11
+revision=1
+wrksrc="${pkgname}-${version}-c"
+build_style=gnu-configure
+hostmakedepends="pkg-config automake"
+makedepends="libev-devel cairo-devel pam-devel libxkbcommon-devel xcb-util-devel xcb-util-image-devel libjpeg-turbo-devel"
```
Comment (maxice8) [inline srcpkgs/i3lock-color/template]: please cut this line at 80 columns and prefix the following with an whitespace.

**Comment by maxice8** [comment]
please add the following to srcpkgs/i3lock-color/patches/<name>.patch

change <name> to a fitting name, i reccomend fix-musl

```
From c265fea7039300ff2283446ce6e7bc03439e2860 Mon Sep 17 00:00:00 2001
From: maxice8 <thinkabit.ukim@gmail.com>
Date: Wed, 27 Jun 2018 21:55:47 -0300
Subject: [PATCH] jpg.h: add include for sys/types.h and define _GNU_SOURCE.

this allows us to use uint on musl libc systems like Void Linux and
Alpine.
---
 jpg.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/jpg.h b/jpg.h
index b1146bc..fb74e78 100644
--- a/jpg.h
+++ b/jpg.h
@@ -1,6 +1,9 @@
 #ifndef _JPG_H
 #define _JPG_H
 
+#include <sys/types.h>
+
+#define _GNU_SOURCE 1
 typedef struct {
     uint height;
     uint width;
```

File: srcpkgs/i3lock-color/template [hunk 8]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'i3lock-color'
+pkgname=i3lock-color
+version=2.11
+revision=1
+wrksrc="${pkgname}-${version}-c"
+build_style=gnu-configure
+hostmakedepends="pkg-config automake"
+makedepends="libev-devel cairo-devel pam-devel libxkbcommon-devel xcb-util-devel
+	xcb-util-image-devel libjpeg-turbo-devel"
```
Comment (maxice8) [inline srcpkgs/i3lock-color/template]: should be one whitespace, not 1 tab.

see template with multiple lines like gtk+3 to see how it is done.

**Comment by jtalowell** [comment]
@maxice8 I've corrected the template. Let me know if there's anything else that needs changing.

**Pull request merged**